### PR TITLE
Auth context

### PR DESCRIPTION
**Pull request by bajajcodes**
## Auth context for login and signup pages
 - Login page uses auth context to make user login into account and change status to logged-in as true 
   using helper methods imported from `main.helpers.js`
- Signup page uses auth context to create user account and set status for signup as true and change status 
   for logged-in true as well, with help of helpers method imported from `main.helpers.js`
- Helpers methods in `main.helpers.js` provides components `Login and Signup` control to manage states and 
  handle user input validation on the component itself and send status for successful login or signup to auth context.

## What files to review
- **src/context/auth/**
- **src/pages/auth/main/**

## Suggestion for review
- Use of validation logic, storing of encoded token and handling of validation by components is correct.

## Preview
 
 - Signup

https://user-images.githubusercontent.com/29247011/160367822-afe98ab0-6c06-401e-a3e3-f25669be958a.mp4

- Login

https://user-images.githubusercontent.com/29247011/160367895-fd2612ea-4194-47a7-8590-24d47b3264ba.mp4

### CODE REVIEW

**Review by lannisterrr — APPROVED** [review]
Really clean code , Learned about how to save user information in local storage.

File: src/pages/auth/main/main.signup.js [hunk 1]
```diff
@@ -1,57 +1,157 @@
...
       }));
+
+      if (isSignuped) {
+        authDispatch({
+          type: "SIGNUP",
+        });
+        navigate("/products");
+      }
+
+      if (isSignuped === false) {
```
Comment (prachi-sahani) [inline src/pages/auth/main/main.signup.js]: can be combined in if-else condition

File: src/pages/auth/main/main.login.js [hunk 2]
```diff
@@ -1,27 +1,82 @@
...
+    setPassword("");
+
+    if (loggedIn) {
+      authDispatch({
+        type: "LOGIN",
+      });
+      navigate("/products");
+    }
+
+    if (loggedIn === false) {
```
Comment (prachi-sahani) [inline src/pages/auth/main/main.login.js]: can be written as if-else condition

File: src/pages/auth/alert/alert.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  return (
+    <div
+      className="alert alert-bg-error dflex align-items-start"
+      style={{ display: displayValue ? "-webkit-box" : "none" }}
+    >
+      <span className="material-icons alert-info-icon alert-color-error mr-9">
+        cancel
+      </span>
+      <div>
+        <div className="alert-message">{`${message}.`}</div>
```
Comment (prachi-sahani) [inline src/pages/auth/alert/alert.js]: {`${message}.`} can be simply written as {message}

**Review by prachi-sahani — COMMENTED** [review]
the code is understandable and good to merge

suggestions
 - code can be more composed and formatted
 - {" "} can be removed

File: src/pages/auth/alert/alert.js (hunk 3 above)
Comment (bajajcodes) [reply src/pages/auth/alert/alert.js]: Done.

File: src/pages/auth/main/main.login.js (hunk 2 above)
Comment (bajajcodes) [reply src/pages/auth/main/main.login.js]: Applied.

File: src/pages/auth/main/main.signup.js (hunk 1 above)
Comment (bajajcodes) [reply src/pages/auth/main/main.signup.js]: Applied.

**Pull request merged**